proto/private/rules/proto_toolchain.bzl

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A toolchain for Protobuf rules."""
+
+load("//proto/private:native.bzl", proto_common = "native_proto_common")
+
+def _proto_toolchain_impl(ctx):
+    return [
+        platform_common.ToolchainInfo(
+            protoc = ctx.attr._protoc[DefaultInfo].files_to_run,
+            protoc_options = getattr(ctx.fragments.proto, "protoc_options", []),
```

Comment:
So how Java does it, you can set flags on the toolchain as an attribute, and you can override them with a flag. In Java rules the toolchain exposes the attribute and actual rules implementation make the override from the flag.

It is not a bad idea to do the override in the toolchain already. Perhaps consider adding protoc_opts attribute to the rule as well.